pkg/kubectl/rolling_updater.go

```diff
@@ -501,9 +503,16 @@ func (r *RollingUpdater) cleanupWithClients(oldRc, newRc *api.ReplicationControl
 		return err
 	}
 
-	if err = wait.Poll(config.Interval, config.Timeout, client.ControllerHasDesiredReplicas(r.c, newRc)); err != nil {
+	watchOptions := api.ListOptions{FieldSelector: fields.OneTermEqualSelector("metadata.name", newRc.Name), ResourceVersion: "0"}
```

Comment:
Create a helper to build the `ListOptions`, `SingleResourceWatchOptions` or something like that.